src/diffusers/pipelines/controlnet_xs/pipeline_controlnet_xs.py

```diff
@@ -139,6 +189,49 @@ def __init__(
         )
         self.register_to_config(requires_safety_checker=requires_safety_checker)
 
+    def get_base_pipeline(self):
+        """Get underlying `StableDiffusionPipeline` without the `ControlNetXSAddon` model."""
+        components = {k: v for k, v in self.components.items() if k != "controlnet"}
+        components["unet"] = self.components["controlnet"].base_model
+        return StableDiffusionPipeline(**components)
+
+    def get_controlnet_addon(self):
+        """Get the `ControlNetXSAddon` model."""
+        return self.components["controlnet"].ctrl_addon
```

Comment:
why do we need this?

Reply:
We don't. It's outdated. Have deleted it.